app/src/main/java/acr/browser/lightning/view/LightningView.java

```diff
@@ -581,14 +587,38 @@ private String getMobileUserAgent() {
         final String defaultUserAgent = mWebView.getSettings().getUserAgentString();
         final Matcher matcher = USER_AGENT_PATTERN.matcher(defaultUserAgent);
         final String userAgent;
-        if (matcher.matches() && matcher.groupCount() >= 5) {
-            userAgent = matcher.group(1) + matcher.group(2) + matcher.group(5);
+        if (matcher.matches() && matcher.groupCount() >= 3) {
+            final String product = matcher.group(1);
+            final String systemAndPlatform = matcher.group(2);
+            final String extensions = matcher.group(3);
+            final String cleanSystemAndPlatform = getCleanSystemAndPlatform(systemAndPlatform);
+            userAgent = String.format(
+                    "%s (%s) %s Cliqz/%s",
+                    product,
+                    cleanSystemAndPlatform,
+                    extensions,
+                    BuildConfig.VERSION_NAME
```

Comment:
Is there any good reason to include version in User Agent? Is version name just a product name or it include actual product version?

Reply:
The format is <product>/<version>, to me version can be a generic 1.0.0